ext/scheduler/airflow/airflow.go

```diff
@@ -284,8 +281,29 @@ func (s *Scheduler) GetJobRuns(ctx context.Context, tnnt tenant.Tenant, jobQuery
 	if err := json.Unmarshal(resp, &dagRunList); err != nil {
 		return nil, errors.Wrap(EntityAirflow, fmt.Sprintf("json error on parsing airflow dag runs: %s", string(resp)), err)
 	}
+	return &dagRunList, nil
+}
 
-	return getJobRuns(dagRunList, jobCron)
+func (s *Scheduler) GetJobRuns(ctx context.Context, tnnt tenant.Tenant, jobQuery *scheduler.JobRunsCriteria, jobCron *cron.ScheduleSpec) ([]*scheduler.JobRunStatus, error) {
+	spanCtx, span := startChildSpan(ctx, "GetJobRuns")
+	defer span.End()
+
+	dagRunListResp, err := s.getDagRunList(spanCtx, tnnt, jobQuery, jobCron)
+	if err != nil {
+		return nil, err
+	}
+	return getJobRuns(*dagRunListResp, jobCron, true)
+}
+
+func (s *Scheduler) GetScheduledJobRuns(ctx context.Context, tnnt tenant.Tenant, jobQuery *scheduler.JobRunsCriteria, jobCron *cron.ScheduleSpec) ([]*scheduler.JobRunStatus, error) {
+	spanCtx, span := startChildSpan(ctx, "GetScheduledJobRuns")
+	defer span.End()
+
+	dagRunListResp, err := s.getDagRunList(spanCtx, tnnt, jobQuery, jobCron)
+	if err != nil {
+		return nil, err
+	}
+	return getJobRuns(*dagRunListResp, jobCron, false)
```

Comment:
this flag can be better put up in `scheduler.JobRunsCriteria` this will require very minimal code change , and also avoid the need for code duplication.

Reply:
was thinking about this previously, agree with that. revert [this commit](https://github.com/goto/optimus/pull/114/commits/fa473b2a9c20639427fb8dd7365dc7dbacc3a635) and added new commit for adding withExternalTrigger flag on jobrunscriteria